woocommerce/woocommerce-android
# UI of the IPP feedback survey banner

#### UI of the IPP feedback survey banner

**Note:** This PR is part of the IPP in-app feedback banner project https://github.com/woocommerce/woocommerce-android/issues/8116. It targets the feature branch feature/8116-ipp-feedback-banner.

### Description


This PR adds the UI part of the IPP feedback banner.

Details:
* `FeedbackSurveyFragment` was updated to allow passing custom URLs instead of `SurveyType` enum.
* Implemented the banner under the `FeedbackBanner` class and `ipp_feedback_banner.xml`  layout.
* Added banner displaying logic to `OrderListFragment` and `OrderListViewModel` classes.
* Displaying dismiss dialog when "X" button is clicked on the banner.
* Added `FeatureFlag.IPP_FEEDBACK_BANNER` which is checked in view model before displaying the banner

### Testing instructions


1. Verify that the IPP banner is displayed on the Orders screen. Conditions: COD enabled, WCPay plugin enabled.
2. Verify that the "Create orders" expandable banner is not shown when IPP banner is visible.
3. Verify that the survey is opened when "Share feedback" button is clicked.
4. Verify that the "Create orders" expandable banner is shown again when IPP banner is hidden (e.g. dismissed, or survey is opened)
5. Verify dismiss dialog is shown when clicking "X" button.

**Note:** The final user group detection conditions will be adjusted in the next PR. Please skip testing if the correct banner message and title are shown now.

### Roadmap
In the next PRs, I'll update the user group detection conditions and add tracking events.

### Images/gif

| IPP banner hidden  | IPP banner shown |
| ------------- | ------------- |
| ![Screenshot_20230119-130907](https://user-images.githubusercontent.com/4527432/213440177-f0a11921-2e09-48d4-8302-5d06ebca9822.jpg)  | ![Screenshot_20230119-131051](https://user-images.githubusercontent.com/4527432/213440195-7599aaa4-f90f-4532-8151-75366b121e4a.jpg)  |
